src/lib/pay-request/services/connector/client.ts

```diff
@@ -49,7 +49,7 @@ export default class Connector extends Client {
             return client._axios
                 .get(`/v1/api/accounts/${accountId}/charges/${externalChargeId}`)
                 .then(response => client._unpackResponseData<Charge>(response))
-                .catch(handleEntityNotFound("Charge", externalChargeId))
+                .catch(handleChargeNotFoundForParityCheck("Charge", externalChargeId, true))
```

Comment:
I think it would be simpler/easier to understand if you just introduced a new convenience method. 

Something like `parityCheck(externalChargeId: string ...)` could call the same endpoint but just handle a 404 response differently without the need to pass in conditional bools

Reply:
fixed